lib/engine.js

```diff
@@ -57,7 +57,30 @@ function pickStorage() {
   return inMemoryStore;
 }
 
+function isReadSupportedNatively() {
```

Comment:
In theory this is not needed if the prior `isSupportedNatively` is `true` but wasnt keen on leaking information between the two tests

Reply:
Fixed in the last commit where we now have these isSupportedNatively functions storing the result in a var to be used